File: ecommerce-shoe-shop-frontend/src/app/common/order-item.ts
import { CartItem } from "./cart-item";

export class OrderItem {
    part_price: number;
    quantity: number;
    product_id: number;

    constructor(cartItem: CartItem){
        this.product_id = cartItem.id;
        this.quantity = cartItem.quantity;
        this.part_price = cartItem.price;
    }
}

File: Spring-Boot-e-commerce-shoe-shop/src/main/java/com/Soft_Dev_Project/SpringBootecommerceshoeshop/entity/OrderItem.java
package com.Soft_Dev_Project.SpringBootecommerceshoeshop.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Entity
@Table(name = "orderitem")
@Getter
@Setter
@ToString
public class OrderItem {

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Column(name = "id")
        private Long id;

        private Long product_id;

        private int quantity;

        private int part_price;

        private int size;

        @ManyToOne
        @JoinColumn(name = "order_id")
        private Order order;

}
